backend/src/test/java/com/team4/backend/service/InternshipOfferServiceTest.java

```diff
@@ -354,4 +358,110 @@ void shouldGetNotYetValidatedInternshipOffer() {
+
+        when(studentService.findByEmail(any(String.class))).thenReturn(Mono.just(student));
+        when(studentService.addOfferToStudentAppliedOffers(any(Student.class), any(String.class)))
+                .then(s -> {
+                    student.getAppliedOffersId().add(internshipOffer.getId());
+                    return Mono.just(student);
+                });
+
+        int sizeBefore = InternshipOfferMockData.getInternshipOffer().getListEmailInterestedStudents().size();
+
+        //ACT
+        Mono<InternshipOffer> internshipOfferMono = internshipOfferService.applyOffer(
+                internshipOffer.getId(),
+                principal
+        );
+
+        //ASSERT
+        StepVerifier.create(internshipOfferMono)
+                .assertNext(offer -> assertEquals(sizeBefore + 1, offer.getListEmailInterestedStudents().size()))
+                .verifyComplete();
+    }
+
+    @Test
+    void shouldNotApplyInternshipOfferInvalidStudent() {
+        //ARRANGE
+        InternshipOffer internshipOffer = InternshipOfferMockData.getInternshipOffer();
+        Student student = StudentMockData.getMockStudent();
+        internshipOffer.getListEmailInterestedStudents().add(student.getEmail());
+
+        Principal principal = mock(Principal.class);
+        when(UserSessionService.getLoggedUserEmail(principal)).thenReturn(student.getEmail());
+        when(internshipOfferRepository.findById(internshipOffer.getId())).thenReturn(Mono.just(internshipOffer));
+        when(studentService.findByEmail(student.getEmail())).thenReturn(Mono.error(UserNotFoundException::new));
+
+        //ACT
+        Mono<InternshipOffer> internshipOfferMono = internshipOfferService.applyOffer(
+                internshipOffer.getId(),
+                principal
+        );
+
+        //ASSERT
+        StepVerifier.create(internshipOfferMono)
+                .expectError(UserNotFoundException.class)
+                .verify();
+    }
+
+    @Test
+    void shouldNotApplyInternshipOfferInvalidOffer() {
+        //ARRANGE
+        InternshipOffer internshipOffer = InternshipOfferMockData.getInternshipOffer();
+        Student student = StudentMockData.getMockStudent();
+        internshipOffer.getListEmailInterestedStudents().add(student.getEmail());
+
+        Principal principal = mock(Principal.class);
+        when(internshipOfferRepository.findById(internshipOffer.getId()))
+                .thenReturn(Mono.error(InternshipOfferNotFoundException::new));
+
+        //ACT
+        Mono<InternshipOffer> internshipOfferMono = internshipOfferService.applyOffer(
+                internshipOffer.getId(),
+                principal
+        );
+
+        //ASSERT
+        StepVerifier.create(internshipOfferMono)
+                .expectError(InternshipOfferNotFoundException.class)
+                .verify();
+    }
+
+    @Test
+    void shouldNotApplyInternshipOfferNotInExclusiveStudents() {
```

Comment:
the name of this test should be reflect what it does when i read it it is not clear to me what case it is testing, it should shouldNotApplyToInternshipOfferNotIncludedInStudentExclusiveOffers or something like that